cmake/parthenonConfig.cmake.in

```diff
@@ -0,0 +1,26 @@
+#=========================================================================================
+# (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+#
+# This program was produced under U.S. Government contract 89233218CNA000001 for Los
+# Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+# for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+# in the program are reserved by Triad National Security, LLC, and the U.S. Department
+# of Energy/National Nuclear Security Administration. The Government is granted for
+# itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+# license in this material to reproduce, prepare derivative works, distribute copies to
+# the public, perform publicly and display publicly, and to permit others to do so.
+#=========================================================================================
+
+# Favor using the kokkos package that was built with parthenon, if one has not been specified 
+if ( NOT Kokkos_FOUND )
+  find_package(Kokkos 3 REQUIRED PATHS @CMAKE_INSTALL_PREFIX@)
+  set(Kokkos_BUILT_WITH_CUDA @Kokkos_ENABLE_CUDA@)
+  if(${Kokkos_BUILT_WITH_CUDA})
+    string(FIND ${CMAKE_CXX_COMPILER} "nvcc_wrapper" FOUND_WRAPPER)
+    if( ${FOUND_WRAPPER} EQUAL "-1" ) 
+      message(WARNING "Kokkos was built with cuda, recommend setting CMAKE_CXX_COMPILER to @CMAKE_INSTALL_PREFIX@/bin/nvcc_wrapper")
```

Comment:
In case of an external kokkos `@CMAKE_INSTALL_PREFIX@/bin/nvcc_wrapper` isn't right.

Reply:
@junghans I agree that this isn't right for an external kokkos installation but what variable should we use instead of `CMAKE_INSTALL_PREFIX`? I see that [`INTERFACE` libraries lack a `LOCATION`](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#interface-libraries); the closest I can get from cmake is the location of the Kokkos include directory.

I would like to try something similar with replacing `CMAKE_INSTALL_PREFIX` in the `find_package(Kokkos ...` call above and adding `NO_DEFAULT_PATH` since my understanding is that manually re-adding Kokkos as a library here would require a lot of typing since Kokkos is composed of a few targets (`Kokkos::kokkoscontainers`, etc.) that is already handled by KokkosConfig.cmake

Reply:
Why aren't you setting a variable like `EXTERNAL_KOKKOS`, then you could use `@EXTERNAL_KOKKOS@` in here.

Reply:
Yes this works, thanks. I changed around the logic in parthenon/CMakeLists.txt so if `EXTERNAL_KOKKOS` exists it is preferred even if `external/Kokkos` or `Kokkos_ROOT` exists.